### PR TITLE
feat: add file.rmdir

### LINKED ISSUE #1267: Extend file module functions (for smoother GitOps)
Hello there!

KCL being a very capable versatile language should have more file system control functions:
  - mkdir
  - delete
 And maybe even more than this.
 
It's essential to have control over transitional file creation/deletion and placement. For instance in my case, I do generate prometheus rules from parameterized KCL definitions as the first step:

```
cd appsops/rules/graph && mkdir -p rules manifests && kcl -dY ./kcl.yaml
generate_rules:
- graph/rules/common-certmanager.yaml
- graph/rules/common-elasticsearch.yaml
- graph/rules/common-istio.yaml
- graph/rules/common-k8s-events.yaml
- graph/rules/common-k8s-nodes.yaml
- graph/rules/common-k8s-resources.yaml
- graph/rules/common-vector.yaml
```

This is the configuration entry for the next stage, which proceeds with VMRule manifests generation. Simply there should be enough control in place on where to place the files, how to create or delete them to _enable smooth GitOps experience_ out of box. In fact KCL can be and should be used independently to provide a solid GitOps experience.

### PR DESCRIPTION
**Pull request by JeevaRamanathan**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 
Fixes #1267 

#### 2. What is the scope of this PR (e.g. component or file name):
kcl-lang/kcl/kclvm

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

### CODE REVIEW

File: test/grammar/builtins/file/rmdir/stdout.golden [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (Peefy) [inline test/grammar/builtins/file/rmdir/stdout.golden:1]: Remove empty stdout file.

**Comment by JeevaRamanathan** [comment]
@Peefy is there a way to sign-off specific commit [added file rmdir](https://github.com/kcl-lang/kcl/pull/1289/commits/725901e65afaf252f1b6b688d3aba4bbabbffd2b) [725901e], due to which dco is failing.

**Comment by Peefy** [comment]
> @Peefy is there a way to sign-off specific commit [added file rmdir](https://github.com/kcl-lang/kcl/pull/1289/commits/725901e65afaf252f1b6b688d3aba4bbabbffd2b) [[725901e](https://github.com/kcl-lang/kcl/commit/725901e65afaf252f1b6b688d3aba4bbabbffd2b)], due to which dco is failing.

See https://github.com/kcl-lang/kcl/pull/1289/checks?check_run_id=24665928148

File: test/grammar/builtins/file/rmdir/stdout.golden (hunk 1 above)
Comment (JeevaRamanathan) [reply test/grammar/builtins/file/rmdir/stdout.golden:1]: I have removed it!

Comment (Peefy) [reply test/grammar/builtins/file/rmdir/stdout.golden:1]: Remove the test/grammar/builtins/file/rmdir/stdout.golden ...

**Comment by Peefy** [comment]
The CI failed and please rebase the main branch.

**Comment by Peefy** [comment]
The CI failed and please rebase the main branch and squash the commit.
image

**Comment by Peefy** [comment]
It should be noted that this implementation conflicts with the existing implementation file. delete function, so I will close this PR. https://www.kcl-lang.io/docs/reference/model/file#delete

**Pull request closed without merging**